File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int n;
    cin >> n;
    int tri[501][501];
    int dp[501][501] = {0};

    for(int i = 1; i <= n; i++) {
        for(int j = 1; j <= i; j++) {
            cin >> tri[i][j];
        }
    }

    dp[1][1] = tri[1][1];
    for(int i = 2; i <= n; i++) {
        for(int j = 1; j <= i; j++) {
            dp[i][j] = max(dp[i-1][j-1], dp[i-1][j]) + tri[i][j];
        }
    }
    int max = 0;
    for(int i = 1; i <= n; i++) {
        if(max < dp[n][i])
        max = dp[n][i];
    }
    cout << max;
}

File: 백준/Silver/1932. 정수 삼각형/정수 삼각형.py
tri = []
n = int(input())
for i in range(n):
    tri.append(list(map(int, input().split())))


for i in range(1, n):
     for j in range(len(tri[i])):
        # 가장 왼쪽 또는 가장 오른쪽인지 확인
        if j == 0:
            tri[i][j] += tri[i-1][j]
        elif j == (len(tri[i]) - 1):
             tri[i][j] += tri[i-1][j-1]
        else:
            tri[i][j] = max(tri[i][j] + tri[i-1][j-1], tri[i][j] + tri[i-1][j])

print(max(tri[n-1]))